### ISSUE TITLE
Problem with tracking part

### ISSUE THREAD

**Issue by LiXm1002** [VisualComputingInstitute/TrackR-CNN#23]
Thanks for sharing.
I am interested in Hungarian algorithm mentioned in the paper. Which python file is the algorithm wrote in? I didn't find it.
Thanks for someone helping me.

**Comment by LiXm1002** [comment]
@pvoigtlaender  Hi, when I train on my own dataset, loss is decreasing,but when I test the data in training dataset , there are no prediction on the picture,Is my training time not long enough? I think the loss is very low.
272 / 3688 {loss:  0.25727}  2.1365833282470703
  280 / 3688 {loss:  0.16748}  1.989861249923706
  288 / 3688 {loss:  0.23351} elapsed 1.8858006000518799
  296 / 3688 {loss:  0.26398} elapsed 2.1159489154815674
  304 / 3688 {loss:  0.17822} elapsed 2.0314927101135254
  312 / 3688 {loss:  0.22028} elapsed 2.071747303009033
  320 / 3688 {loss:  0.23223} elapsed 2.1517374515533447

**Comment by pvoigtlaender** [comment]
elapsed just means how much time the step took, so it is expected not to drop but to stay roughly constant :)

**Comment by pvoigtlaender** [comment]
you can find the Hungarian algorithm here:
https://github.com/VisualComputingInstitute/TrackR-CNN/blob/master/forwarding/tracking/Util_tracking.py#L169

**Comment by LiXm1002** [comment]
Thanks！And I have made a small dataset by myself（About 500 images, two videos, it’s too difficult to label the dataset.T_T）, Its scene is very different from KITTI and MOTChallenge , Tracking targets are no longer cars and pedestrians. GPU is 2080Ti ,How long does it take to train?

**Comment by pvoigtlaender** [comment]
yeah, pixel-wise labeling for each frame is indeed lots of effort..
For the training, expect a few days. For KITTI we trained ~4 days I think. Since your dataset is smaller, maybe you can train a bit shorter.

**Comment by LiXm1002** [comment]
Oh，thanks！I hope to get a good result.

**Comment by LiXm1002** [comment]
@pvoigtlaender Hi,When use Hungarian algorithm to match detection,Is it calculates the detection of the (t-1) frame and the t frame ,or prediction of the t-1 frame at the t frame comparing with detection on t frame?

**Comment by pvoigtlaender** [comment]
What exactly do you mean with prediction?

We take the current active tracks from before time-steps (so in most cases this will be detections from t-1) and match them against time t.
There is no prediction step like done by a Kalman filter if that is what you mean

**Comment by LiXm1002** [comment]
So the Hungarian algorithm matches the target detection between different frames.There is no prediction step in it.I understood wrong before.
